src-ghc/Pact/Server/PactService.hs

```diff
@@ -91,18 +93,67 @@ runPayload c@Command{..} = do
-
-applyContinuation :: ContMsg -> [UserSig] -> CommandM p CommandResult
-applyContinuation _ _ = throwCmdEx "Continuation not supported"
+  CommandState {..} <- liftIO $ readMVar _ceState
+  let sigs = userSigsToPactKeySet _cmdSigs
+      evalEnv = setupEvalEnv _ceDbEnv _ceEntity _ceMode
+                (MsgData sigs edata Nothing _cmdHash) _csRefStore
+  EvalResult{..} <- liftIO $ evalExec evalEnv parsedCode
+  newPact <- join <$> mapM (handleYield erInput _cmdSigs) erExec
+  let newState = CommandState erRefStore $ case newPact of
+        Nothing -> _csPacts
+        Just (cp@CommandPact{..}) -> M.insert _cpTxId cp _csPacts
+  void $ liftIO $ swapMVar _ceState newState
+  return $ jsonResult _ceMode rk $ CommandSuccess (last erOutput)
+
+-- Better name?
+handleYield :: [Term Name] -> [UserSig] -> PactExec -> CommandM p (Maybe CommandPact)
+handleYield em cmdSigs PactExec{..} = do
+  CommandEnv{..} <- ask
+  --TODO: handle entity?
+  unless (length em == 1) $
+    throwCmdEx $ "handleYield: defpact execution must occur as a single command: " ++ show em
+  case _ceMode of
+    Local -> return Nothing
+    Transactional tid -> do
+      --TODO: handle yielded objects
+      let sigs = userSigsToPactKeySet cmdSigs
+      return $ Just $ CommandPact tid (head em) sigs _peStepCount _peStep
+
+
+applyContinuation :: RequestKey -> ContMsg -> [UserSig] -> Hash -> CommandM p CommandResult
+applyContinuation rk ContMsg{..} cmdSigs cmdHash = do
+  CommandEnv{..} <- ask
+  case _ceMode of
+    Local -> throwCmdEx "Local continuation exec not supported"
+    Transactional _ -> do
+      CommandState{..} <- liftIO $ readMVar _ceState
+      case M.lookup _cmTxId _csPacts of
+        Nothing -> throwCmdEx $ "applyContinuation: txid not found: " ++ show _cmTxId
+        Just CommandPact{..} -> do
+          let nextStep = _cpStep + 1
+              isLast = nextStep >= _cpStepCount
+              
+          when (isLast) $ throwCmdEx $ "applyContinuation: reaping pact [disabled]: " ++ show _cmTxId
+          when (_cmStep < 0 || _cmStep >= _cpStepCount) $ throwCmdEx $ "Invalid step value: " ++ show _cmStep
+          when (_cmStep /= nextStep) $ throwCmdEx $ "Invalid step value: Received " ++ show _cmStep ++ " but expected " ++ show nextStep
+          
+          let sigs = userSigsToPactKeySet cmdSigs
+              pactStep = Just $ PactStep nextStep _cmRollback (PactId $ pack $ show $ _cmTxId) Nothing -- TODO Resume
+              evalEnv = setupEvalEnv _ceDbEnv _ceEntity _ceMode
+                (MsgData sigs Null pactStep cmdHash) _csRefStore   -- TODO data from msg
+          
+          EvalResult{..} <- liftIO $ evalContinuation evalEnv _cpContinuation
+          
+          let updatePactState stepNum = CommandPact _cpTxId _cpContinuation _cpSigs _cpStepCount stepNum
+              newState = CommandState erRefStore $ case erExec of
+                Nothing -> _csPacts --TODO does this ever occur?
```

Comment:
hmm, seems like an impossibility, but would imply that the `_cpContinuation` didn't run a pact, so should blow up with dire error